web/modules/custom/asset_release/src/Guard/AssetReleaseFulfillmentGuard.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Drupal\asset_release\Guard;
+
+use Drupal\asset_release\AssetReleaseRelations;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\joinup_user\WorkflowUserProvider;
+use Drupal\og\Og;
+use Drupal\rdf_entity\RdfInterface;
+use Drupal\state_machine\Guard\GuardInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowInterface;
+use Drupal\state_machine\Plugin\Workflow\WorkflowTransition;
+
+/**
+ * Guard class for the transitions of the asset release entity.
+ *
+ * @package Drupal\asset_release\Guard
+ */
+class AssetReleaseFulfillmentGuard implements GuardInterface {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  private $entityTypeManager;
+
+  /**
+   * Holds the workflow user object needed for the checks.
+   *
+   * This will almost always return the logged in users but in case a check is
+   * needed to be done on a different account, it should be possible.
+   *
+   * @var \Drupal\joinup_user\WorkflowUserProvider
+   */
+  private $workflowUserProvider;
+
+  /**
+   * The asset release relation manager.
+   *
+   * @var \Drupal\asset_release\AssetReleaseRelations
+   */
+  private $assetReleaseRelationManager;
+
+  /**
+   * Instantiates a AssetReleaseFulfillmentGuard service.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *    The WorkflowUserProvider service.
+   * @param \Drupal\joinup_user\WorkflowUserProvider $workflow_user_provider
+   *    The WorkflowUserProvider service.
+   * @param \Drupal\asset_release\AssetReleaseRelations $asset_release_relations
+   *    The Asset release relation service.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, WorkflowUserProvider $workflow_user_provider, AssetReleaseRelations $asset_release_relations) {
```

Comment:
Same as below but that is a good suggestion indeed and already considered.
Though, can we skip it for the next workflow? the reason is that this ticket is already too much loaded.